community/cypher/compatibility-suite/src/test/scala/cypher/feature/steps/CypherTCKSteps.scala

```diff
@@ -154,6 +164,23 @@ class CypherTCKSteps extends FunSuiteLike with Matchers with TCKCucumberTemplate
     }
   }
 
+  private def checkcompiletimeError(result: Try[Result], status: String, phase: String, detail: String): Unit = {
+    result match {
+      case Failure(e: QueryExecutionException) =>
+        s"Neo.ClientError.Statement.$status" should equal(e.getStatusCode)
+
+        if (e.getMessage.matches("Invalid input .+ is not a valid value, must be a positive integer[\\s.\\S]+"))
+          detail should equal("LiteralMustBePositiveInteger")
+        else fail(s"Unknown compile time error: $e")
+
+      case Failure(e) =>
+        fail(s"Unknown compile time error: $e")
+
+      case _: Success[_] =>
+        fail("No compile time error was raised")
```

Comment:
I think this function could share its code with the other one (`checkruntimeErrror`), and have the string name (the only differentiator that I can see) be sent as a parameter instead.